packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -1517,6 +1505,7 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
     // Essentially the same as a standard OUT_OF_GAS, except we also retain a record of the gas
     // refund to be given at the end of the transaction.
     if (messageRecord.nuisanceGasLeft < _amount) {
+        messageRecord.nuisanceGasLeft = 0;
```

Comment:
ref: https://linear.app/optimism/issue/OP-658/oz-m05-nuisance-gas-left-is-not-reduced-to-zero-when-operation-exceeds

Reply:
Also address this one: https://linear.app/optimism/issue/OP-653/oz-h05-repeatedly-exceeding-nuisance-gas-limit

Reply:
Might also need this: https://linear.app/optimism/issue/OP-950/provide-a-minimum-nuisance-gas-budget-to-call-frames